src/main/java/org/everit/jira/analytics/event/NoEstimateUsageChangedEvent.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright (C) 2011 Everit Kft. (http://www.everit.org)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.everit.jira.analytics.event;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.regex.Pattern;
+
+import org.everit.jira.analytics.PiwikUrlBuilder;
+import org.everit.jira.timetracker.plugin.JiraTimetrackerAnalytics;
+import org.everit.jira.timetracker.plugin.util.PiwikPropertiesUtil;
+
+/**
+ * Non-estimated field changed event.
+ */
+public class NoEstimateUsageChangedEvent implements AnalyticsEvent {
+  /**
+   * Enumeration for representing the three option for the non estimate field.
+   */
+  private enum NonEstUsageValues {
```

Comment:
Tell me some reason why? I think this is the most encapsulated solution. There is no other class which need to know anything about this class.